pytorch_lightning/utilities/auto_restart.py

```diff
@@ -70,31 +70,37 @@ def worker_id(self) -> int:
 
     def __iter__(self) -> Iterator[Any]:
         self._current_iteration = 0
+        self.i = 0
+        self.sampler_iter = iter(self._sampler)
+        return self
+
+    def __next__(self):
```

Comment:
Note for the reviewer: This is used to simplify state management and don't change the FastForwardSampler.